### PR TITLE
Adds CommonJS/RequireJS support

### PR DESCRIPTION
**Pull request by christopherjohst**
This adds support for asynchronous module definitions (AMD) through both RequireJS and CommonJS. If neither is present, the plugin is added to the global jQuery.

### CODE REVIEW

File: jquery.ajaxchimp.js [hunk 1]
```diff
@@ -25,7 +25,15 @@ To get the mailchimp JSONP url (undocumented), change 'post?' to 'post-json?' an
 For e.g. 'http://blahblah.us1.list-manage.com/subscribe/post-json?u=5afsdhfuhdsiufdba6f8802&id=4djhfdsh99f&c=?',
 */
 
-(function ($) {
+;(function (factory) {
+    if (typeof define === 'function' && define.amd) {
+        define(['jquery'], factory);
+    } else if (typeof exports === 'object') {
```
Comment (scdoshi) [inline jquery.ajaxchimp.js:31]: not very familiar with requireJS, so I may be way off base here, but should this not be 

```
else if (typeof module === 'object' && module && module.exports) {
```

Comment (scdoshi) [inline jquery.ajaxchimp.js:31]: nvm. seems both would work based on this: https://github.com/substack/browserify-handbook#exports

**Comment by scdoshi** [comment]
will be covered in v2.0: #57

**Comment by christopherjohst** [comment]
Just saw this, yes they both do the same thing. `module.exports` is commonJS syntax though, `define` is for requireJS.

**Pull request closed without merging**